packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -5705,13 +5728,14 @@ abstract class RenderObjectElement extends Element {
 
   @override
   void unmount() {
+    final RenderObjectWidget oldWidget = widget;
     super.unmount();
     assert(
       !renderObject.attached,
       'A RenderObject was still attached when attempting to unmount its '
       'RenderObjectElement: $renderObject',
     );
-    widget.didUnmountRenderObject(renderObject);
+    oldWidget.didUnmountRenderObject(renderObject);
```

Comment:
Is there a good reason to not null the `renderObject` here?

Reply:
That should be fine.